docs/specifications/tests/DNSSEC-TP/dnssec16.md

```diff
@@ -106,40 +115,52 @@ DS16_MIXED_DELETE_CDS                | ERROR   | "Delete" CDS record is mixed wi
 
-    1. If the CDS RRset is non-empty, get the RRset and its RRSIG
-       records, else go to next name server IP address. The set of the RRSIG
-       records may be empty.
-    2. If any CDS record is a "delete" CDS, then do:
+    1. If the CDS RRset is empty go to next name server IP address.
+    2. Get the CDS RRset and its RRSIG records from the *CDS RRsets* set
+       for the name server IP. The RRSIG records may be absent.
+    3. If any CDS record is a "delete" CDS, then do:
        1. If there is more than a single CDS record then add the name
           server IP to the *Mixed Delete CDS* set.
        2. Else, add the name server IP address to the *Delete CDS*
           set.
-       3. Go to next name server IP.
-    3. Get the DNSKEY and its RRSIG records from the *DNSKEY RRsets* for the same
+    4. Get the DNSKEY and its RRSIG records from the *DNSKEY RRsets* for the same
        name server IP. The set of RRSIG records may be empty.
-    4. If there are no DNSKEY records, then do:
+    5. If there are no DNSKEY records, then do:
        1. Add name server IP address to the *No DNSKEY RRset* set.
        2. Go to next name server IP.
-    5. Repeat the following steps for each CDS record:
+    6. Repeat the following steps for each CDS record unless it is a "delete"
+       CDS record:
        1. Compare the key tag from the CDS record with the calculated
           key tags for the DNSKEY records.
        2. If the CDS record does not match any DNSKEY record then add
           the name server IP address and CDS record key tag to the
           *No Match CDS With DNSKEY* set.
-       3. Else, if there is no RRSIG for the DNSKEY RRset created by
-          the DNSKEY record that the CDS record points at then add the
-          name server IP address and key tag of CDS record to the
-          *DNSKEY Not Signed By CDS* set.
-    6. If there are no RRSIG records for the CDS RRset, then add the
+       3. Else, if bit 7 of the flag field of the DNSKEY that the DS record
+          points to is set to 0 (nil) then add the name server IP address and
+          CDS record key tag to the *CDS points to non-zone DNSKEY* set.
+       4. Else, do:
+          *  If there is no RRSIG for the DNSKEY RRset created by the DNSKEY
+             record that the CDS record points at then add the name server IP
+             address and key tag of CDS record to the
+             *DNSKEY Not Signed By CDS* set.
+          *  If there is no RRSIG for the CDS RRset created by the DNSKEY
+             record that the CDS record points at then add the name server IP
+             address and key tag of CDS record to the *CDS Not Signed By CDS*
+             set.
+          *  If bit 15 of the flag field of the DNSKEY that the DS record
+             points to is set to 0 (nil) then add the name server IP address
+             and the key tag of the CDS record to the
+             *CDS points to non-SEP DNSKEY* set.
```

Comment:
There's no corresponding step for this in DNSSEC17. Is that intentional?

Reply:
Yes, look for "CDNSKEY is non-SEP key". When you have a CDS you have to find the corresponding DNSKEY to determine if the SEP is set or not. With CDNSKEY you at look at the record itself.